server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -1951,13 +1952,21 @@ public class DateFormatters {
     /////////////////////////////////////////
 
     static DateFormatter forPattern(String input) {
+        return forPattern(input, Version.CURRENT);
+    }
+
+    static DateFormatter forPattern(String input, Version supportedVersion) {
         if (Strings.hasLength(input)) {
             input = input.trim();
         }
         if (input == null || input.length() == 0) {
             throw new IllegalArgumentException("No date pattern provided");
         }
 
+        if (supportedVersion.major < Version.V_8_0_0.major) {
```

Comment:
I wonder if this wouldn't fit better in DateFormatter. We already do some other bwc changes in DateFormatter class
any particular reason you preferred it here?

Reply:
No particular reason, I can move it up in the DateFormatter. Thanks.